### DISCUSSION TITLE
`clang-format` and ranges

### DISCUSSION THREAD

**Discussion by kylechui** [clangd/clangd#1445] [Q&A]
Is there a way to format the pipe operator that's used in C++20 ranges to show one per line? For example:
```cpp
// what currently happens
auto view = someVar | views::drop(sz / 2)
            | views::filter([&](auto c) { return half.contains(c); })
            | views::common;
// what I would like
auto view = someVar
            | views::drop(sz / 2)
            | views::filter([&](auto c) { return half.contains(c); })
            | views::common;
```

**Comment by HighCommander4** [comment]
Maybe the range-v3 library's [.clang-format](https://github.com/ericniebler/range-v3/blob/master/.clang-format) file contains some useful options?

If not, a new option for this may need to be added to clang-format. You can file an issue in the [LLVM issue tracker](https://github.com/llvm/llvm-project/issues) for it, and once it's implemented clangd will automatically start to respect it when used.
